File: Day2/strong_number.py
""" Program for the Strong Number"""
def get_factorial(n):
    """
    Calculates the factorial of a digits in a given number.
    """
    prod=1
    while n>0:
        prod *= n
        n = n-1
    return prod

def get_is_strong_number(n):
    """Checks if a given number is an strong number.
    """
    temp=n
    total_sum = 0
    while temp>0:
        digit = temp % 10
        total_sum +=  get_factorial(digit)
        temp //= 10
    return num == total_sum

num = int(input("Enter the number : "))
ans = get_is_strong_number(num)
if ans:
    print(num,"is a Strong Number. ")
else:
    print(num,"is not a Strong Number.")

File: Day2/StrongNumber.cpp
#include<iostream>
using namespace std;

int getFactorial(int n){
    int factorial = 1;
    while(n != 0){
        factorial *= n;
        n--;
    }
    return factorial;
}
bool getIsArmStrongNumber(int num){
    int temp = num;
    int sum =0;
    while(num > 0){
        int digit = num % 10;
        sum += getFactorial(digit);
        num /= 10;
    }
    return temp == sum;
}
int main(){

    int num;
    cout<<"Enter the Number: ";
    cin>>num;

    bool ans = getIsArmStrongNumber(num);
    if (ans==true)
    {
        cout<<num<<" is a Strong Number.";
    }
    else{
        cout<<num<<" is not a Strong Number.";
    }
    
}
